src/main/java/org/jenkinsci/plugins/github/config/GitHubPluginConfig.java

```diff
@@ -171,7 +171,7 @@ public boolean configure(StaplerRequest req, JSONObject json) throws FormExcepti
 
     @Override
     public String getDisplayName() {
-        return "GitHub Plugin Configuration";
+        return "GitHub Configuration";
```

Comment:
Github plugin provides connection for plugins, it has own documentation that guides on how configure this connection. I.e. in pullrequest-plugin i have reference: 1) configure github-plugin 2) configure PR plugin. 
Please remember that non enterprise usage it's not marketing feature where you want mask everything and resolve all issues via support. FOSS users should understand what and where should be configured. 

I think we can review and bit update github-plugin documentation to indicate that it could be used by other plugins.Technically it has API + ancient push->GitSCM.poll() parts. Other plugins should also reference to github-plugin (i.e. if configured user should have additional permissions).